config.go

```diff
@@ -116,20 +154,78 @@ func PrivateKey(key *rsa.PrivateKey) Option {
 	}
 }
 
+// PrivateKeyFile sets the RSA private key in the secure channel configuration
+// from a PEM or DER encoded file.
+func PrivateKeyFile(keyFile string) Option {
```

Comment:
I like the idea of supporting multiple filetypes and this is probably the best way to handle these helper functions.  That means, however, we should have one for `Certficate`, too.

Users need to provide both a cert and a key in every case so the two config options should feel the same.